src/libsync/networkjobs.cpp

```diff
@@ -554,6 +555,42 @@ bool CheckServerJob::finished()
 
 /*********************************************************************************************/
 
+CheckRedirectCostFreeUrlJob::CheckRedirectCostFreeUrlJob(AccountPtr account, QObject *parent)
+    : AbstractNetworkJob(account, QLatin1String(statusphpC), parent)
+{
+    setIgnoreCredentialFailure(true);
+}
+
+void CheckRedirectCostFreeUrlJob::start()
+{
+    setFollowRedirects(false);
+    sendRequest("GET", Utility::concatUrlPath(account()->url(), QStringLiteral("/index.php/204")));
+    AbstractNetworkJob::start();
+}
+
+void CheckRedirectCostFreeUrlJob::onTimedOut()
+{
+    qCWarning(lcCheckRedirectCostFreeUrlJob) << "TIMEOUT";
+    if (reply() && reply()->isRunning()) {
+        emit timeout(reply()->url());
+    } else if (!reply()) {
+        qCWarning(lcCheckRedirectCostFreeUrlJob) << "Timeout even there was no reply?";
+    }
+    deleteLater();
```

Comment:
will not this cause issue for the existing termination code ?
maybe just replace this by a call to `AbstractNetworkJob::onTimedOut` ?